agent/exploits/cve_2024_21287.py

```diff
@@ -0,0 +1,33 @@
+"""Agent Asteroid implementation for CVE-2024-21287"""
+
+import re
+
+from packaging import version
+
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+VULNERABILITY_TITLE = "Oracle Agile PLM Framework Remote File Disclosure"
+VULNERABILITY_REFERENCE = "CVE-2024-21287"
+VULNERABILITY_DESCRIPTION = """A vulnerability in Oracle Agile Product Lifecycle Management (PLM) 
+was discovered, allowing remote attackers to exploit a file disclosure issue. This vulnerability 
+can be exploited over the network without authentication, potentially disclosing sensitive files."""
+RISK_RATING = "HIGH"
+MAX_VULNERABLE_VERSION = version.parse("9.3.6")
+VERSION_PATTERN = re.compile(r"Build Number:\s*(\d+\.\d+\.\d+)")
+
+
+@exploits_registry.register
+class CVE202421287Exploit(webexploit.WebExploit):
+    accept_request = definitions.Request(method="GET", path="/Agile/")
+    check_request = definitions.Request(method="GET", path="/Agile/")
+    accept_pattern = [re.compile("Agile Product Lifecycle Management")]
+    vuln_ranges = [definitions.VulnRange(None, MAX_VULNERABLE_VERSION)]
+    metadata = definitions.VulnerabilityMetadata(
+        title=VULNERABILITY_TITLE,
+        description=VULNERABILITY_DESCRIPTION,
+        reference=VULNERABILITY_REFERENCE,
+        risk_rating=RISK_RATING,
+    )
+    version_pattern = VERSION_PATTERN
```

Comment:
Question : do we don't need the check_pattern or this version pattern do the job